src/extractor/guidance/intersection_generator.cpp

```diff
@@ -176,7 +176,7 @@ Intersection IntersectionGenerator::mergeSegregatedRoads(Intersection intersecti
         const auto &first_data = node_based_graph.GetEdgeData(intersection[first].turn.eid);
         const auto &second_data = node_based_graph.GetEdgeData(intersection[second].turn.eid);
 
-        return first_data.name_id != INVALID_NAME_ID && first_data.name_id == second_data.name_id &&
+        return first_data.name_id != INVALID_NAMEID && first_data.name_id == second_data.name_id &&
```

Comment:
The new INVALID_NAMEID which you changed (your PR has to be merged first) is `0`. So INVALID_NAME_ID was zero and will, after your PR also be 0 again. Right now it is broken.